packages/core/js-client/src/fetchers/browser.ts

```diff
@@ -15,26 +15,47 @@
  */
 
 interface PackageJsonContent {
-    dependencies: Record<string, string | undefined>;
-    devDependencies: Record<string, string | undefined>;
+  dependencies: Record<string, string | undefined>;
+  devDependencies: Record<string, string | undefined>;
 }
 
 // This will be substituted in build phase
 const packageJsonContentString = `__PACKAGE_JSON_CONTENT__`;
-let parsedPackageJsonContent: PackageJsonContent;
+let parsedPackageJsonContent: PackageJsonContent | undefined;
 
-const PRIMARY_CDN = "https://unpkg.com/"; 
+const PRIMARY_CDN = "https://unpkg.com/";
 
 export async function fetchResource(pkg: string, assetPath: string) {
-    const packageJsonContent = parsedPackageJsonContent || (parsedPackageJsonContent = JSON.parse(packageJsonContentString));
-    const version = packageJsonContent.dependencies[pkg] || packageJsonContent.devDependencies[pkg];
-    
-    if (version === undefined) {
-        const availableDeps = [...Object.keys(packageJsonContent.dependencies), ...Object.keys(packageJsonContent.devDependencies)];
-        throw new Error(`Cannot find version of ${pkg} in package.json. Available versions: ${availableDeps.join(',')}`);
-    }
-    
-    const refinedAssetPath = assetPath.startsWith('/') ? assetPath.slice(1) : assetPath;
-    
-    return fetch(new globalThis.URL(`${pkg}@${version}/` + refinedAssetPath, PRIMARY_CDN));
+  const packageJsonContent =
+    parsedPackageJsonContent ??
+    // TODO: Should be validated
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    (parsedPackageJsonContent = JSON.parse(
+      packageJsonContentString,
+    ) as PackageJsonContent);
```

Comment:
all this is known at compile time. so replacing on build step, or file reading and parsing and validating - all can be avoided if file is just imported. Yeah it will include it one more time, but I think it's simpler and not that bad, because when bundeled - it will be bundeled only once. And for nodejs package size just doesn't matter

Reply:
This is already bundled as a string. It looks ok to me.
But i tried your approach and for some reasons rootDir becomes 1 folder behind and adds `src` surrounding to `dist` folder. Manually restricting rootDir to src folder wont work.

Reply:
I changed this to build source files with package.json. Please review

Reply:
Well, after this change, `pnpm install` went mad and cant understand where is the real package.json.
So i had to revert this.
I think to achieve typesafe approach we need to copy this file first, give it different name, probably take only subset of it and use that subset.
In this case i prefer options with keeping the content as a string which is simpler in my opinion. Probably we can reconsider this later.